### PR TITLE
Rework "Jump to Code" buttons to use data from routine annotations

### PR DESCRIPTION
**Pull request by markerikson**
> **NOTE**: This PR depends on https://github.com/replayio/backend/pull/7759 landing to fix issues with the event listeners routine, and then we need that routine turned on for real and running consistently before we can switch from "figure out the location when the user clicks" to "use the precalculated location data from the routine".

This PR:

- Converts the existing annotation handling for React DevTools, Redux DevTools, and annotation kinds into single-entry caches
  - deletes the existing `state.reactDevTools` slice
  - Updates the TS types and some variable names in `ReactDevTools.tsx` to better describe the parsed annotation contents
- Updates the info events panel and Cypress test steps panelto use precalculated annotations data for the "Jump" buttons

This doesn't yet delete the existing client-side "jump" logic - I can do that in a follow-up PR.

### CODE REVIEW

**Comment by bvaughn** [comment]
This is still titled "DRAFT" so I'll hold off on reviewing it.

**Comment by markerikson** [comment]
FWIW the logic looks good atm, so I think the code is ready for review. I just don't want to merge it until I've had a further chance to confirm that it all works right with real annotations data coming from the backend.  Logan and I tried to check that a couple hours ago and it looked okay at first glance, but the backend was doing a revert and we had to set the checking aside.

File: src/ui/setup/dynamic/devtools.ts [hunk 1]
```diff
@@ -212,10 +214,16 @@ export default async function setupDevtools(store: AppStore, replayClient: Repla
...
   setupNetwork(store, replayClient);
-  setupReactDevTools(store, ThreadFront);
   setupBoxModel(store, startAppListening);
   setupRules(store, startAppListening);
 
+  ThreadFront.waitForSession().then(() => {
+    // Precache annotations
+    annotationKindsCache.readAsync(replayClient);
+    reactDevToolsAnnotationsCache.readAsync();
+    eventListenersJumpLocationsCache.readAsync();
```
Comment (bvaughn) [inline src/ui/setup/dynamic/devtools.ts:224]: Trivial naming nit: `cache.prefetch` might be a better semantic fit

**Review by bvaughn — APPROVED** [review]
I don't have a ton of context to this code so I just skimmed it. Looks good though.

**Review by bvaughn — APPROVED** [review]
I don't have a ton of context to this code so I just skimmed it. Looks good though.

**Pull request merged**